tests/clients/test_job.py

```diff
@@ -291,6 +292,11 @@ def test_job_definition_schema():
     assert data == JOB_DEFINITION
 
 
+def validate_job_definition_schema():
```

Comment:
Functions need to start with `test_` to be detected as test by `pytest` - I think if you look at the logs when running the tests that this will not appear.

I think that our convention for a test of this sort would be to name it `test_job_definition_schema_invalid`.

It would be good if we could write a separate test for the other invalid case (when the instanceSizeType is `custom` and the `instanceSize` is unset. You could then write either a separate test for this case or try parameterising this existing test with `pytest.mark.parametrize` (see elsewhere in this file for an example).